JavaScript/JavaScript.sublime-syntax

```diff
@@ -1057,7 +1060,28 @@ contexts:
         - match: '\}'
           scope: punctuation.section.block.js
           pop: true
-        - include: comma-separator
+
+        - match: >-
+            (?x)(?=
+              {{method_name}}\s*:
+              (?: {{func_lookahead}} | {{arrow_func_lookahead}} )
+            )
+          push:
+            - - meta_scope: meta.function.declaration.js
```

Comment:
Putting this meta scope here means the `meta.function.declaration.js` gets added to the function body as well - and causes duplicate scoping as well. This can be seen by examining the scopes in the following example:
```javascript
var test = {
    a: x => x * 2,
    b: function() {}
}
```